test/server/http_filter_integration_test_base.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <string>
+
+#include "external/envoy/test/integration/http_integration.h"
+
+namespace Nighthawk {
+
+/**
+ * Base class with shared functionality for testing Nighthawk test server http filter extensions.
+ */
+class HttpFilterIntegrationTestBase : public Envoy::HttpIntegrationTest {
+protected:
+  /**
+   * Indicate the expected response origin.
+   */
+  enum class ResponseOrigin { UPSTREAM, EXTENSION };
+  /**
+   * Construct a new HttpFilterIntegrationTestBase instance.
+   *
+   * @param ip_version Specify the ip version that the integration test server will use to listen
+   * for connections.
+   */
+  HttpFilterIntegrationTestBase(Envoy::Network::Address::IpVersion ip_version);
+
+  /**
+   * We don't override SetUp(): tests using this fixture must call setup() instead.
+   * This is to avoid imposing the need to create a fixture per filter configuration.
+   *
+   * @param config configuration to pass to Envoy::HttpIntegrationTest::config_helper_.addFilter.
```

Comment:
is this configuration for the filter being tested?  If so, can we clarify "filter configuration"